src/redux/sagas/extensions/extensionEnable.ts

```diff
@@ -30,33 +40,56 @@ function* extensionEnable({
     if (!isInitialized && initializationParams) {
       const initParams = modifyParams(initializationParams, payload);
 
+      const additionalChannels: {
+        setUserRolesWithProofs?: Channel;
+      } = {};
+
+      if (neededColonyPermissions.length) {
+        const bytes32Roles = intArrayToBytes32(neededColonyPermissions);
+        additionalChannels.setUserRolesWithProofs = {
+          context: ClientType.ColonyClient,
+          params: [address, Id.RootDomain, bytes32Roles],
+        };
+      }
+
       const {
         channels,
         transactionChannels,
-        transactionChannels: { initialise },
+        transactionChannels: { initialise, setUserRolesWithProofs },
       } = yield setupEnablingGroupTransactions(
         meta.id,
         initParams,
         extensionId,
+        additionalChannels,
       );
 
       const batchKey = 'enableExtensions';
 
-      yield all(
-        Object.keys(transactionChannels).map((channelName) =>
-          createGroupTransaction(
-            transactionChannels[channelName],
-            batchKey,
-            meta,
-            {
-              identifier: colonyAddress,
-              methodName: channelName,
-              ...channels[channelName],
-            },
-          ),
+      const effects = Object.keys(transactionChannels).map((channelName) =>
+        createGroupTransaction(
+          transactionChannels[channelName],
+          batchKey,
+          meta,
+          {
+            identifier: colonyAddress,
+            methodName: channelName,
+            ...channels[channelName],
+          },
         ),
       );
 
+      /* Delay action creation in development, else block ingestor doesn't detect all on-chain events */
+      if (isDev) {
```

Comment:
For context, firing all actions concurrently was causing the block ingestor to "drop" or miss events, and therefore was  setting permissions but not initialising the extension in the db. 

Nothing wrong with colonyjs or interacting with chain itself, this is purely a case of the block ingestor not hearing the event correctly. 

Adding the delay appears to solve the problem, at least on my machine. 

I expect this will cease to be a problem once we change the processing to a block by block basis. ( @jakubcolony )